### PR TITLE
[Feature-14550][datasource-plugin] add HANA datasource-add pom.xml 

### PR DESCRIPTION
**Pull request by IT-Kwj**
add HANA datasource-add pom.xml

### CODE REVIEW

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-hana/pom.xml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        <dependency>
+            <groupId>org.apache.dolphinscheduler</groupId>
+            <artifactId>dolphinscheduler-datasource-api</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>com.sap.cloud.db.jdbc</groupId>
+            <artifactId>ngdbc</artifactId>
+            <version>${ngdbc.version}</version>
```
Comment (SbloodyS) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-hana/pom.xml]: After taking a quick look at Hana's license https://tools.hana.ondemand.com/developer-license-3_1.txt. It doesn't seem to comply with the Apache license.

Comment (zhongjiajie) [reply dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-hana/pom.xml]: 1. could you please add sap jdbc to module bom like and use it in Hana pom https://github.com/apache/dolphinscheduler/blob/2f37da0dbcbbd887a801c6e922551ce561a606cd/dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-mysql/pom.xml#L46
2. and change scope due to not compatible reason

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-all/pom.xml [hunk 2]
```diff
@@ -132,7 +132,10 @@
             <artifactId>dolphinscheduler-datasource-vertica</artifactId>
             <version>${project.version}</version>
         </dependency>
-
+        <dependency>
+            <groupId>org.apache.dolphinscheduler</groupId>
+            <artifactId>dolphinscheduler-datasource-hana</artifactId>
+            <version>${project.version}</version>
```
Comment (zhongjiajie) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-all/pom.xml:138]:
```suggestion
            <version>${project.version}</version>
        </dependency>
```

File: dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-hana/pom.xml [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+            <groupId>com.sap.cloud.db.jdbc</groupId>
+            <artifactId>ngdbc</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>org.mockito</groupId>
+            <artifactId>mockito-core</artifactId>
+            <type>jar</type>
+            <scope>test</scope>
+        </dependency>
```
Comment (zhongjiajie) [inline dolphinscheduler-datasource-plugin/dolphinscheduler-datasource-hana/pom.xml]:
```suggestion
```

**Review by SbloodyS — COMMENTED** [review]
Please resolve conflicts.

**Comment by zhongjiajie** [comment]
close by https://github.com/apache/dolphinscheduler/pull/15260 sorry for replying later, and thanks for your contribution @IT-Kwj

**Pull request closed without merging**